commit 9c5dd59d4b1927b7d88e8e3c2e260eb01d95794b
Author: Vaxry <vaxry@vaxry.net>
Date:   Sat Jun 29 00:23:02 2024 +0200

    input: fix capabilities enum types passed
    
    hyprland down to the seat protocol impl expects IHID capabilities, not WL_ ones
    
    ref #6702 #6196

diff --git a/src/managers/input/InputManager.cpp b/src/managers/input/InputManager.cpp
index 81a46f97..b3621520 100644
--- a/src/managers/input/InputManager.cpp
+++ b/src/managers/input/InputManager.cpp
@@ -1473,14 +1473,7 @@ void CInputManager::updateCapabilities() {
         if (h.expired())
             continue;
 
-        auto cap = h->getCapabilities();
-
-        if (cap & HID_INPUT_CAPABILITY_KEYBOARD)
-            caps |= WL_SEAT_CAPABILITY_KEYBOARD;
-        if (cap & HID_INPUT_CAPABILITY_POINTER)
-            caps |= WL_SEAT_CAPABILITY_POINTER;
-        if (cap & HID_INPUT_CAPABILITY_TOUCH)
-            caps |= WL_SEAT_CAPABILITY_TOUCH;
+        caps |= h->getCapabilities();
     }
 
     g_pSeatManager->updateCapabilities(caps);
